docs/manual/src/idl/interfaces.md

```diff
@@ -0,0 +1,40 @@
+# Interfaces/Objects
+
+Interfaces refer to the Rust world as a struct with an `impl` block containing methods. In the Kotlin or Swift world, it's a class.  
+Because of Reasons, in the uniffi world it is impossible to be both an Object and a [Dictionary](./structs.md).
```

Comment:
If we wanted to be slightly less mysterious about The Reasons, we could describe objects are being passed around by reference while dictionaries are passed around by value (or by copy or similar).